test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -363,14 +370,45 @@ public void afterEach(ExtensionContext context) throws Exception {
         }
     }
 
+    // We need the usual ClassLoader hacks in order to present the callbacks with the proper test object and context
     private Map.Entry<Class<?>, ?> createQuarkusTestMethodContextTuple(ExtensionContext context) throws Exception {
+        ClassLoader classLoader = runningQuarkusApplication.getClassLoader();
         if (quarkusTestMethodContextClass == null) {
-            quarkusTestMethodContextClass = Class.forName(QuarkusTestMethodContext.class.getName(), true,
-                    runningQuarkusApplication.getClassLoader());
+            quarkusTestMethodContextClass = Class.forName(QuarkusTestMethodContext.class.getName(), true, classLoader);
         }
+
+        Method originalTestMethod = context.getRequiredTestMethod();
+        Class<?>[] originalParameterTypes = originalTestMethod.getParameterTypes();
+        Method actualTestMethod = null;
+
+        // go up the class hierachy to fetch the proper test method
+        Class<?> c = actualTestClass;
+        while (c != Object.class) {
+            try {
+                List<Class<?>> parameterTypesFromTccl = new ArrayList<>(originalParameterTypes.length);
+                for (Class<?> type : originalParameterTypes) {
```

Comment:
This is done for every class in hierarchy (until the method is found). Wouldn't _once_ be sufficient?

Reply:
You're right! It got moved into the loop when I added the it (the initial implementation didn't have it)

Reply:
> the initial implementation didn't have it

Yeah, I noticed that. So you even fixed three problems! 🎉

Reply:
I didn't set off to do so, that's for sure :)